### PR TITLE
FD - 41

### PR DESCRIPTION
**Pull request by kskulikova**
DELETE badge api endpoint with tests

### CODE REVIEW

File: ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+        [Fact]
+        public async Task DeleteBadgeApi_CorrectAuthentication_ShouldDeleteLevelsAssociatedWithBadge()
+        {
+            var request = new HttpRequestMessage(HttpMethod.Delete, "/api/badges/1");
+            request.Headers.Add("Authorization", "Bearer " + _tokenService.GenerateToken(email, true));
+            var levels = testContext.Context.Badges.GetItemByIndex(0).Levels;
+
+            await testContext.Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            for (var i = 0;
```
Comment (AlexTeper) [inline ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs]: This for loop looks a little bit strange

File: ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        [Fact]
+        public async Task DeleteBadgeApi_CorrectAuthentication_ShouldDeletePitchAssociatedWithBadgeFromUser()
+        {
+            var request = new HttpRequestMessage(HttpMethod.Delete, "/api/badges/1");
+            request.Headers.Add("Authorization", "Bearer " + _tokenService.GenerateToken(email, true));
+            var levels = testContext.Context.Badges.GetItemByIndex(0).Levels;
+            var pitches = levels.SelectMany(l => l.Pitches).ToList();
+
+            await testContext.Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            foreach (var t in pitches)
```
Comment (AlexTeper) [inline ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs]: Please use meaningful variable names

File: ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        public async Task DeleteBadgeApi_CorrectAuthentication_ShouldDeleteReviewAssociatedWithBadgeFromUser()
+        {
+            var request = new HttpRequestMessage(HttpMethod.Delete, "/api/badges/1");
+            request.Headers.Add("Authorization", "Bearer " + _tokenService.GenerateToken(email, true));
+            var levels = testContext.Context.Badges.GetItemByIndex(0).Levels;
+            var pitches = levels.SelectMany(l => l.Pitches).ToList();
+            var reviews = pitches.SelectMany(p => p.Reviews).ToList();
+
+            await testContext.Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            foreach (var t in reviews)
```
Comment (AlexTeper) [inline ferrilata-devilline.IntegrationTests/Scenarios/ApiControllers/Badges/ApiDeleteBadgeTests.cs]: Here too

File: ferrilata-devilline/Services/BadgeService.cs [hunk 4]
```diff
@@ -18,6 +18,12 @@ public BadgeService(IBadgeRepository badgeRepository, IMapper mapper)
             _mapper = mapper;
         }
 
+        public Badge FindBadge(long id)
+        {
+            return _badgeRepository.FindBadgeById(id);
+        }
+
+
```
Comment (AlexTeper) [inline ferrilata-devilline/Services/BadgeService.cs]: Unnecessary  empty line

File: ferrilata-devilline/Services/BadgeService.cs [hunk 5]
```diff
@@ -28,5 +34,15 @@ public List<Badge> GetAll()
         {
             return _badgeRepository.RetrieveBadgesFromDB();
         }
+
+        public void DeleteById(long id)
+        {
+            _badgeRepository.DeleteBadgeById(id);
+        }
+
+        public void SaveorUpdate(long id)
```
Comment (AlexTeper) [inline ferrilata-devilline/Services/BadgeService.cs]: Empty method

File: ferrilata-devilline/Services/BadgeService.cs [hunk 6]
```diff
@@ -18,6 +18,12 @@ public BadgeService(IBadgeRepository badgeRepository, IMapper mapper)
             _mapper = mapper;
         }
 
+        public Badge FindBadge(long id)
+        {
+            return _badgeRepository.FindBadgeById(id);
```
Comment (AlexTeper) [inline ferrilata-devilline/Services/BadgeService.cs:23]: Why is this not async?

**Review by AlexTeper — CHANGES_REQUESTED** [review]

File: ferrilata-devilline/Services/BadgeService.cs (hunk 6 above)
Comment (iaroslavm) [reply ferrilata-devilline/Services/BadgeService.cs:23]: Yes, why not async though?

**Review by iaroslavm — APPROVED** [review]

**Review by AlexTeper — APPROVED** [review]

**Pull request merged**